core/lib/dependency-graph/simulator/network-analyzer.js

```diff
@@ -126,11 +126,17 @@ class NetworkAnalyzer {
 
       // If the request was SSL we get two estimates, one for the SSL negotiation and another for the
       // regular handshake. SSL can also be more than 1 RT but assume False Start was used.
-      if (timing.sslStart > 0 && timing.sslEnd > 0) {
-        return [timing.connectEnd - timing.sslStart, timing.sslStart - timing.connectStart];
-      } else if (timing.connectStart > 0 && timing.connectEnd > 0) {
-        return timing.connectEnd - timing.connectStart;
+      /** @type {number[]} */
+      let estimates = [];
+      if (timing.sslStart >= 0 && timing.sslEnd >= 0) {
+        estimates = [timing.connectEnd - timing.sslStart, timing.sslStart - timing.connectStart];
+      } else if (timing.connectStart >= 0 && timing.connectEnd >= 0) {
+        estimates = [timing.connectEnd - timing.connectStart];
       }
+      // `timing.sslStart - timing.connectStart` can be zero in cases where a TCP conneciton has
+      // already been established but needs to be upgraded to use SSL. Filter out that case, and
```

Comment:
Do you mean reusing an existing connection but now laying SSL on top? I don't think that's done. Or do you mean a racing issue? That's more plausible, but then it runs into an issue with the intention of the `if (connectionReused) return` line above.

The `3700.147` request in `amp-m86.devtoolslog.json` log is notably over h3. Chrome will (usually) optimistically open an h3 connection in parallel to an h2 connection in case the h3 fails, which could be the case here (there's an h2 request (`3700.12`) to the same `fonts.googleapis.com` origin a second earlier), so in that case by the time `3700.147` is made the connection is already partially established, which could be the reason for the partial timing.

_But_, it's also h3, which doesn't have a TCP handshake in the first place :)

Reply:
> Do you mean reusing an existing connection but now laying SSL on top? I don't think that's done. Or do you mean a racing issue? That's more plausible, but then it runs into an issue with the intention of the if (connectionReused) return line above.

Racing issue then, I suppose.

FYI there is a non-h3 example of this being zero for `node core/scripts/lantern/debug-url.js https://www.tianya.cn/m/`